TemplePlus/dispatcher.cpp

```diff
@@ -72,6 +77,40 @@ int DispatcherReplacements::Dispatch54AoE(objHndl handle, DispIO* evtObj, D20Dis
 	return 0;
 }
 
+int DispatcherReplacements::GetNumNaturalAttacks_(objHndl handle) {
+	if (!handle) return 0;
+	auto obj = objSystem->GetObject(handle);
+	int result = 0;
+	int idx = 0;
+	int attacksCount;
+	do
+	{
```

Comment:
do{} while is bad form. It's an artifact of the decompiler, there's no need to adhere to it.